firmware/quintuna/RSM/src/io/io_rPump.c

```
@@ -0,0 +1,44 @@
+#include "hw_i2cs.h"
+
+#define MAX_WIPER_VALUE (256.0f)
+#define MIN_WIPER_VALUE (0.0f)
+
+bool io_rPump_isPumpReady(const I2cDevice *device)
+{
+    return hw_i2c_isTargetReady(device);
+}
+
+void io_rPump_write(const I2cDevice *device, uint8_t data)
+{
+    uint8_t buffer[1];
+    buffer[0] = data;
+    hw_i2c_memoryWrite(device, 0x02, buffer, sizeof(buffer));
+}
+
+uint8_t io_rPump_read(const I2cDevice *device)
+{
+    uint8_t buffer[1];
+    hw_i2c_memoryRead(device, 0x02, buffer, sizeof(buffer));
+    return buffer[0];
+}
```

Specify in the function names what you are writing/reading, also hold `0x02` in a macro with a name indicative of what it represents...

Additionally, you don't need to specify a buffer array here, for example for read, you can also do:

```c
    uint8_t res;
    hw_i2c_memoryRead(device, 0x02, &res, sizeof(res));
    return res;
```

Also memoryRead and memoryWrite can fail, they return false if they do. Please make this function return a success/failure boolean - instead of returning the value directly, write to a pointer provided by the user, ie. for read,

```c
bool io_rPump_read(const I2cDevice *device, uint8_t *dest)
{
    return hw_i2c_memoryRead(device, 0x02, dest, sizeof(uint8_t));
}
```

---

Write can be as simple as:
```c
bool io_rPump_write(const I2cDevice *device, uint8_t input) {
    return hw_i2c_memoryWrite(device, 0x02, &input, sizeof(input));
}
```